### DISCUSSION TITLE
zagraniczne pobrania

### DISCUSSION THREAD

**Discussion by ciaglok2** [allegro/allegro-api#7658] [Suggestions - resolved]
Jedne zagraniczne pobraniowe metody dostawy wymagają podania additionalServices / CashOnDelivery, inne nie.
Wśród tych pierwszych niektóre wymagają numeru rachunku, inne nie.
Wśród nich niektóre wymagają "PL" przed numerem (np. DPD), inne nie.
Packeta i Allegro Czechy DHL wymagają danych o pobraniu, ale nr rachunku musi być pusty.

Miarka się przebrała po wprowadzeniu Allegro International.
Musiałem stworzyć takich reguł kilkanaście.
I odrzucacie taki request zamiast zignorować nadmiarowe dane.

Potrzebna jest standaryzacja

**Comment by hahyes** [comment]
@ciaglok2 Sprawdzałeś, jak to działa na tych nowych zasobach które wypuścili wczoraj? Może tam temat jest lepiej ogarnięty.
https://github.com/allegro/allegro-api/issues/7656

**Comment by MartaNowaczyk** [comment]
@ciaglok2 
Co do standaryzacji IBAN i przedrostków PL, jest to indywidualny wymóg przewoźnika i Allegro nie ma wpływu na jego postać - dlatego tutaj niestety nie wprowadzimy zmian.
Co do kwestii wymagalności IBAN, to w nowym zasobie [GET /shipment-management/delivery-services](https://developer.allegro.pl/documentation/#operation/getDeliveryServices) mamy pole, które "mówi" wprost czy podać IBAN czy nie. Mowa o polu `services.cashOnDelivery.forceRequireIban`. 
Sugestia częściowo wdrożona, dlatego oznaczam jako rozwiązaną.
